### PR TITLE
small simple bug-fix: update lost infos in function create_dataset function

### PR DESCRIPTION
**Pull request by im-Kitsch**
# Description

This is fix for simple bug-fix for function `create_dataset_from_collector_env()`.  
1. Add docstring for parameters
2. Add metadata attribute to dataset, meta data like `algorithm name` `author` `author Email` is not saved though it's passed into function.

for function `create_dataset_from_collector_env()`
ref_min/ref_max score is not saved to metadata, it's easily to find the problem.

I think the bugs are quite clear and definitely is a bug, you can take a look,

## Type of change

- Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `pre-commit run --all-files` (see `CONTRIBUTING.md` instructions to set it up)
- [x] I have run `pytest -v` and no errors are present.
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I solved any possible warnings that `pytest -v` has generated that are related to my code to the best of my knowledge.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by balisujohn** [comment]
Good catches. Thanks for your contribution! All these fixes look correct to me.  I think it would be good to make the values for `author_name`, `author_email`, `code_permalink`  default to `"not provided"` instead of `str(None)` if not provided. Once that's done I will merge this PR after a final review.  As for minari version as an argument, I'm wondering if we should make it non-optional, but we can sort that out in a future PR.

**Comment by im-Kitsch** [comment]
Hi, 

I pushed update to save `author_name`, `author_email`, `code_permalink` and `algorithm name` only when it's not None.  And additionally I changed a code of combine dataset so that it could optional parameters are not saved. 

I agree that we need consider if minari version should be optional or mandatory. 

Futhermore, I think the dataset's structure should be **clearly** specified and let it be **stable**. 

E.g. here in `combine_dataset`, if `author` or `author_email` are not saved then it will throw error. 
https://github.com/Farama-Foundation/Minari/blob/ea4097854cae1b333ddd7de2fc3fabd0fbc1076d/minari/utils.py#L277-L290

There should be still small errors like this one.

File: minari/utils.py
Comment (balisujohn) [inline minari/utils.py:292]: please remove commented out lines

**Review by balisujohn — CHANGES_REQUESTED** [review]
The code seems almost ready to merge. Please remove the commented out lines, and add a basic test to for behavior when the  optional `dataset_metadata` keys modified in this PR are provided when creating an instance of `MinariDataset`

**Comment by im-Kitsch** [comment]
Hi, 
I added one test case for `combine dataset` and modified `create dataset` test case, could you take a look? 

I think these tests should be modified in the future again, since currently those additional infomations could not be got directly from `Minaridataset` instance. But have to be read manualy from h5py file.

File: minari/utils.py
Comment (younik) [inline minari/utils.py:509]: can you remove these comments?

File: tests/utils/test_dataset_combine.py
Comment (younik) [inline tests/utils/test_dataset_combine.py:259]: can you change the permalink to be the following? https://github.com/Farama-Foundation/Minari/blob/main/tests/utils/test_dataset_combine.py

File: tests/utils/test_dataset_combine.py
Comment (younik) [inline tests/utils/test_dataset_combine.py:275]: same here

File: tests/utils/test_dataset_creation.py
Comment (younik) [inline tests/utils/test_dataset_creation.py:79]: same here

File: tests/utils/test_dataset_creation.py
Comment (younik) [inline tests/utils/test_dataset_creation.py:294]: same ehre

File: tests/utils/test_dataset_combine.py
Comment (younik) [inline tests/utils/test_dataset_combine.py:279]: btw, this behavior is a little weird in combine dataset (you just save the attributes of the last)
Anyway, it is fixed in https://github.com/Farama-Foundation/Minari/pull/133 that we will merge after, so it is okay for now

**Review by younik — CHANGES_REQUESTED** [review]

File: tests/utils/test_dataset_combine.py
Comment (im-Kitsch) [reply tests/utils/test_dataset_combine.py:279]: Ah, yes, I don't think only saving the attribute of last is a good choice, but it's just to keep same with `combine_dataset()` implementation. Definitely it will be changed in the future, but this PR is just to save the fogortten metadata `code_permalink` and `algorithm_name`.

**Review by younik — APPROVED** [review]
Just fix the pre-commit, and then looks good to me

**Comment by im-Kitsch** [comment]
> Just fix the pre-commit, and then looks good to me

Hi, I think the pre-commit is fixed now.

**Pull request merged**
